### ISSUE TITLE
Can jsdelivr regain its ICP filing in mainland China?

### ISSUE THREAD

**Issue by BNQDZJ** [jsdelivr/jsdelivr#18407]
**Is your feature request related to a problem? Please describe.**
At present, the ICP record of jsdelivr has been revoked, and it has gradually started to be attacked by GFW's DNS pollution and TCP connection reset, resulting in users in mainland China unable to use jsdelivr's services normally.

**Describe the solution you'd like**
Re-acquire ICP filing and use CDN services in mainland China to ensure normal use of users in mainland China

**Describe alternatives you've considered**
Try changing any domain name except .io and .app

**Additional context**
- Some of the content (including this sentence) is machine translated and may not be smooth.
- I think it is important for jsdelivr to regain ICP filing and start using CDN services in mainland China. There is no CDN provided by technology companies in mainland China that can replace jsdelivr.
- Finally, I hope jsdelivr can go further and further on the road of localization in mainland China

**Comment by TestEngineerFish** [comment]
pod install

**Comment by TestEngineerFish** [comment]
use pod install error:
[!] CDN: trunk URL couldn't be downloaded: https://cdn.jsdelivr.net/cocoa/Specs/d/c/2/sqlite3/3.38.5/sqlite3.podspec.json Response: SSL connect error

**Comment by Morax-xyc** [comment]
Please!
There’s no CDN service in China is better than jsdelivr! If jsdelivr can't regain the icp record, I need replace almost all my .js files so that my website works properly (in fact, I have already done so on my blog.)

**Comment by jimaek** [comment]
Unfortunately there isn't anything we can do about it. We maintained a China specific block list but it still wasn't enough apparently. If someone knows how to contact the relevant Chinese organs to discuss the unban of the service please let us know.

In the meantime you can try a service like https://github.com/PipecraftNet/jsdelivr-auto-fallback

**Comment by BNQDZJ** [comment]
> Unfortunately there isn't anything we can do about it. We maintained a China specific block list but it still wasn't enough apparently. If someone knows how to contact the relevant Chinese organs to discuss the unban of the service please let us know.
> 
> In the meantime you can try a service like https://github.com/PipecraftNet/jsdelivr-auto-fallback

The Chinese government currently has no contact information to lift the GFW ban. Only by pointing the domain name to the server in mainland China and filing the ICP record can the ban be lifted.

The envisaged solution: Register a domain name serving mainland China with a domain name registrar in mainland China, and perform ICP filing. When the domain name is blocked, they will remind in time to facilitate rectification. However, these registrars usually need to purchase their services in order to perform ICP filing. You can choose the cheapest service to purchase (Note: Before purchasing the service, please check the documents provided by the service provider in detail to confirm whether the service can be used for filing)

Tip: Tip: Domain name registrars usually provide tutorials with ICP filing, which detail the materials required for filing

It should be noted that this method is not applicable to jsdelivr.net because if you do ICP filing with a domain name registrar in mainland China, they will ask you to change the registrar of the domain name to them, which will cause a lot of problems

I don't know if the above solution is feasible because there are too many Simplified Chinese and Mainland Chinese laws involved, but this is the only solution for now

**Comment by jimaek** [comment]
Even if we register a new domain in China and get the ICP license for it we will still have problems:

1. The source of content is still the same. We don't know why jsDelivr was blocked, and that means the new domain will get blocked as well sooner or later
2. We can't redirect Chinese users to a new domain because the existing one is broken. So they would have to manually make the change.

:(

**Comment by BNQDZJ** [comment]
> Even if we register a new domain in China and get the ICP license for it we will still have problems:
> 
> 1. The source of content is still the same. We don't know why jsDelivr was blocked, and that means the new domain will get blocked as well sooner or later
> 2. We can't redirect Chinese users to a new domain because the existing one is broken. So they would have to manually make the change.
> 
> :(

When a new domain name is about to be blocked Chinese mainland's domain name registrar will notify you in advance, and perhaps you can know why it is blocked through the domain name registrar's notification

When a new domain name appears Chinese mainland users replace the old link themselves, because Chinese mainland users can follow jsdelivr through many channels

**Comment by jimaek** [comment]
>When a new domain name is about to be blocked Chinese mainland's domain name registrar will notify you in advance, and perhaps you can know why it is blocked through the domain name registrar's notification

We also need a Chinese citizen to do this for us and I believe it's a high risk as he would be personally be responsible for any content on jsDelivr. So "wait until we get blocked again so that the government would tell you what we did wrong" is probably too risky.

The ideal scenario is to somehow get information from the Chinese government about the ban so that we could first remove the content they dislike and then get the main domain unblocked.

**Comment by BNQDZJ** [comment]
> > When a new domain name is about to be blocked Chinese mainland's domain name registrar will notify you in advance, and perhaps you can know why it is blocked through the domain name registrar's notification
> 
> We also need a Chinese citizen to do this for us and I believe it's a high risk as he would be personally be responsible for any content on jsDelivr. So "wait until we get blocked again so that the government would tell you what we did wrong" is probably too risky.
> 
> The ideal scenario is to somehow get information from the Chinese government about the ban so that we could first remove the content they dislike and then get the main domain unblocked.

The government will give you a correction period of 7 to 14 days, will not block your domain name immediately, you will not be responsible for completing the correction during this period, and the government will not hold you accountable

Getting information about the ban from the Chinese government (www.gov.cn) requires real-time attention to the laws enacted by the Chinese government and existing laws.

https://cloud.tencent.com/document/product/243 Here are detailed procedures and related legal terms for filing ICP filings in China, but it is Chinese and may require translation software, and websites will not be blocked as long as they comply with the legal terms mentioned there

If you can't access or understand, here's a list of what the Chinese government prohibits: adult content, gambling, drug dealing, gun trading, political dissent, violations of any Chinese laws and policies

What the Chinese government allows but needs to report in advance:
1. Cultural type website

The cultural type website mentioned here refers to the proliferation of cultural products through the network manufacturing, such as audio-visual products, games, animations and other cultural products, such enterprises need to apply to the local cultural department for a business license.

2. Publishing type website

This does not refer to the publishing house, but to the act of editing and uploading one's own or other people's works to the Internet for public reading, and this type of website needs to apply to the relevant departments for an Internet Publishing License.

3. News and information websites

If your website is to do real-time news information release, such as economic, political, diplomatic and other reports and comments, you need to do pre-approval of the website to the relevant departments, and you need to apply for a license from the relevant local departments to be approved to go online.

4. Educational websites

Websites engaged in education training or consulting services must apply to the local education information center for approval before they can go online if they want to be open for business.

5. Pharmaceutical or medical device type website

The state is very strict in the management of medicine and medical devices, and enterprises engaged in the operation of such websites need to do preliminary approval and apply for service review consent from the local health department.

**Comment by Lemmingh** [comment]
[^state-council-292]: [《互联网信息服务管理办法》（国务院令 第 292 号）](<http://www.gov.cn/gongbao/content/2000/content_60531.htm>)

[^state-council-746]: [《中华人民共和国市场主体登记管理条例》（国务院令 第 746 号）](<http://www.gov.cn/gongbao/content/2021/content_5636139.htm>)

[^mii-33]: [《非经营性互联网信息服务备案管理办法》（信息产业部令 第 33 号）](<http://www.gov.cn/gongbao/content/2005/content_93018.htm>)

[^miit-xgj-2009-371]: [《关于建立境内违法互联网站黑名单管理制度的通知》（工信部联电管 〔2009〕371 号）](<https://beian.miit.gov.cn/cms_files/filemanager/431831678/attach/202112/f8be987d88dd4a6f900566f0fd138971.pdf>)

[^company-law]: [《中华人民共和国公司法》](<http://www.npc.gov.cn/npc/c12435/201811/68a85058b4c843d1a938420a77da14b4.shtml>)

[^foreign-investment-law]: [《中华人民共和国外商投资法》](<http://www.npc.gov.cn/npc/c30834/201903/121916e4943f416b8b0ea12e0714d683.shtml>)

[^cloudflare-baidu-partnership]: [How We Extended CloudFlare's Performance and Security Into Mainland China](<https://blog.cloudflare.com/how-we-extended-cloudflares-performance-and-security-into-mainland-china/>)

[^evernote-911101085960906163]:
    Evernote used to have a Wholly Foreign-Owned Enterprise in China according to [国家企业信用信息公示系统 (National Enterprise Credit Information Publicity System)](<http://www.gsxt.gov.cn/>):

    > 统一社会信用代码：911101085960906163
    >
    > |     |     |
    > | --- | --- |
    > | 名称     | 艾沃诺特（北京）科技有限公司 |
    > | (Name)   | EVERNOTE (BEIJING) TECHNOLOGIES CO.,LTD |
    > | 类型     | 有限责任公司（外国法人独资） |
    > | (Type)   | LLC (Foreign) |
    > | 状态     | 注销（决议解散） |
    > | (Status) | Dissolved |



@jimaek

I'd like to echo what @BNQDZJ said above.

Please don't be passive. jsDelivr has to reach the government actively for licenses.

Explicit ICP licenses of your own domains are **mandatory** to operate in China. **Without a license, you are simply illegal**, and doomed to be blocked at any time. [^state-council-292]

---

> the relevant Chinese organs

MIIT (工业和信息化部) is in charge of this. MIIT leads local CAs (通信管理局; communications administration), which issue ICP licenses [^mii-33] , and maintain blacklists [^miit-xgj-2009-371] .

---

> We also need a Chinese citizen to do this for us

You can cooperate with a Chinese CDN provider which can **behave** on behalf of jsDelivr.

To my knowledge, this is how [Cloudflare](<https://www.cloudflare.com/>) made their way to China. [^cloudflare-baidu-partnership]

---

As an alternative solution, you can register a company.

You will need a Chinese lawyer (agent) who can deal with all the tedious stuff on behalf of jsDelivr. There will be a good number of applications and then annual reviews. [^state-council-746] [^company-law] [^foreign-investment-law]

AFAIK, this is how [Evernote](<https://evernote.com/>) used to set up their business in China. [^evernote-911101085960906163]

---

> We don't know why jsDelivr was blocked

After acquiring licenses in China, perhaps jsDelivr can negotiate with the CA that issued your ICP license, or try to contact MIIT directly. Given that jsDelivr is a CDN service, this is arguably necessary to avoid unexpected penalties.

Moreover, the CA will notify you when they decide to enforce a ban. [^miit-xgj-2009-371] Then, you can know what went wrong, and respond in time.

**Comment by jimaek** [comment]
@Lemmingh We had an ICP license until it was revoked for no stated reason. Getting a new one is basically impossible for us.

We also don't speak the language so getting in contact with MIIT is also impossible for us.

We have explored all of these solutions and unfortunately none of them would work in our situation.

**Comment by BNQDZJ** [comment]
The ICP filing may simply be revoked because the ICP filing number is not marked in the footer of the jsdelivr.net

It may also be because jsdelivr has only obtained an ICP filing and not a national Internet security management filing

**Comment by Lemmingh** [comment]
> The ICP filing may simply be revoked because

Additionally, missing the annual review is also a common cause of revocation.

---

> We have explored all of these solutions and unfortunately none of them would work in our situation.

I'm a bit surprised and confused.

Could you please elaborate on why neither Cloudflare nor Evernote's approach applies to jsDelivr?

---

> We also don't speak the language so getting in contact with MIIT is also impossible for us.

I said "Chinese lawyer" and "After acquiring licenses".

---

> Getting a new one is basically impossible for us.

**If you insist on this, as a last resort, please dispel the false hopes of Chinese users.**

Remember, without valid licenses, jsDelivr is now illegal in China. The administration shall block illegal entities as per the regulations I referenced above.

I mean, replace the section <https://www.jsdelivr.com/network#china> with:

> jsDelivr's licenses (<https://github.com/jsdelivr/jsdelivr/issues/729#issuecomment-102442743>) have been revoked. (<https://github.com/jsdelivr/jsdelivr/issues/18348#issuecomment-997777996>)
>
> jsDelivr gives up the Chinese market.
>
> There will be no warranty of jsDelivr service in China.

Thank you for your time.

**Comment by jimaek** [comment]
- We won't transfer our domain to a Chinese domain registrar. This breaks most options to get an ICP license.
- We don't want to create a second domain for Chinese traffic, like cdn.jsdelivr-cn.com. It will completely miss the point of a single unified and global service. Anyone can just mirror jsDelivr to do exactly that. This breaks most other options.
- We don't have the money or resources to hire Chinese law firms to establish Chinese corporations to run our free service
- We plan to update our website in the new redesign to note the revoked ICP license
- We are willing to block any content we need to comply with the local law but we dont know how to get the infringing URLs.

Thanks

**Comment by w0oo** [comment]
> 不幸的是，我们对此无能为力。我们维护了一个针对中国的黑名单，但显然还不够。如果有人知道如何联系中国相关机构讨论解禁服务，请告诉我们。
> 
> 与此同时，您可以尝试像[https://github.com/PipecraftNet/jsdelivr-auto-fallback这样的服务](https://github.com/PipecraftNet/jsdelivr-auto-fallback)

HI, I went to the Chinese Ministry of Industry and Information Technology of the intelligent robot asked a question,
~~lift the damn domain name blacklist only a message, message URL: [http://bzxx.miit.gov.cn/bzxx/appellate/notice](http://bzxx.miit.gov.cn/bzxx/appellate/notice)~~

This is a screenshot 
![Screenshot_2022-08-01-09-12-39-61_40deb401b9ffe8e1df2f1cc5ba480b12.jpg](https://s2.loli.net/2022/08/01/GVy2z46SkBLR9fC.jpg)

# Translate to English
![Screenshot_2022-08-01-09-15-09-63_40deb401b9ffe8e1df2f1cc5ba480b12.jpg](https://s2.loli.net/2022/08/01/aVfjABDbqhGlNIg.jpg)

~~Maybe They can Help You~~

**Comment by BNQDZJ** [comment]
You may try dialing +86 010-66411166, which is the ICP/IP filing contact number provided by the Ministry of Industry and Information Technology

**Comment by CoolPlayLin** [comment]
Hi @jimaek
Maybe you could find another sponsor in China?

**Comment by jimaek** [comment]
We were told that the only way would be to transfer the domain to a Chinese registrar, which is too risky.
If there is an organization willing to help us get the ICP back then without touching the domain then please let me know. Once we have the license, finding a CDN will be simple enough.

**Comment by CoolPlayLin** [comment]
What about cloudflare(China)?

**Comment by jimaek** [comment]
They work though a third party and don't help with ICP

**Comment by Moraxyc** [comment]
> What about cloudflare(China)? That's already sponsor, maybe you could ask him for help?

Cloudflare also needs to obtain an ICP permit in China in order to operate on the Chinese Internet.

**Comment by Jfk90761hg** [comment]
Hola la verdad soy nuevo en esto , jamás e programado pero me gustaría saben que fuentes necesito ver para empezar de 0 , por favor se los agradecería mucho y tengan un exelente día o noche o nos vemos en nuestro sueño.J*1
